### PR TITLE
bug_2410: Allowing Ints to be passed for rows/cols and refactored int…

### LINKED ISSUE #2410: accept int for rows and cols in `fig.add_traces`
At the moment if you want to add several traces to the same subplot you have to pass an iterable to the `rows` and `cols` arguments of `fig.add_traces`. 
```
import plotly.express as px
df = px.data.tips()
fig = px.scatter(df, x='total_bill', y='tip', color='day')
from plotly.subplots import make_subplots
fig2 = make_subplots(1, 2)
fig2.add_traces(fig.data, rows=[1,]*len(fig.data), cols=[1,]*len(fig.data))
fig2.show()
```
We could also accept an int here, in which case all traces would be added to the same subplot.

### PR DESCRIPTION
**Pull request by MCBoarder289**
## Code PR

- [x] I have read through the [contributing notes](https://github.com/plotly/plotly.py/blob/master/contributing.md) and understand the structure of the package. In particular, if my PR modifies code of `plotly.graph_objects`, my modifications concern the `codegen` files and not generated files.
- [x] I have added tests (if submitting a new feature or correcting a bug) or
  modified existing tests.
- [x] For a new feature, I have added documentation examples in an existing or
  new tutorial notebook (please see the doc checklist as well).
- [x] I have added a CHANGELOG entry if fixing/changing/adding anything substantial.

**Changes in this PR:**

- Now allowing the "add_traces" method to take single integers for the rows/cols parameters.
- Also refactored the numpy integer check to be more reusable, and added more tests

Fixes #2410

### CODE REVIEW

File: packages/python/plotly/plotly/basedatatypes.py [hunk 1]
```diff
@@ -1677,14 +1672,19 @@ def add_traces(self, data, rows=None, cols=None, secondary_ys=None):
                   - All remaining properties are passed to the constructor
                     of the specified trace type.
 
-        rows : None or list[int] (default None)
+        rows : None, list[int], or int (default None)
             List of subplot row indexes (starting from 1) for the traces to be
             added. Only valid if figure was created using
             `plotly.tools.make_subplots`
+            If a single integer is added, all traces will be added to row number
```
Comment (emmanuelle) [inline packages/python/plotly/plotly/basedatatypes.py:1679]:
```suggestion
            If a single integer is passed, all traces will be added to row number
```

File: packages/python/plotly/plotly/basedatatypes.py [hunk 2]
```diff
@@ -1677,14 +1672,19 @@ def add_traces(self, data, rows=None, cols=None, secondary_ys=None):
...
+
         cols : None or list[int] (default None)
             List of subplot column indexes (starting from 1) for the traces
             to be added. Only valid if figure was created using
             `plotly.tools.make_subplots`
+            If a single integer is added, all traces will be added to column number
```
Comment (emmanuelle) [inline packages/python/plotly/plotly/basedatatypes.py:1685]:
```suggestion
            If a single integer is passed, all traces will be added to column number
```

**Comment by emmanuelle** [comment]
Thank you very much @MCBoarder289 ! I've left a cosmetic comment about the docstring but this looks ready to me!

**Comment by MCBoarder289** [comment]
Thank you for the review, @emmanuelle! I appreciate the feedback, and those changes are now committed.

**Comment by nicolaskruchten** [comment]
Ah, @emmanuelle I was thinking this would be in 4.9.0 rather than 4.8.2 but OK, it won't break anything for anyone :)

**Pull request merged**